### PR TITLE
Local build instructions

### PR DESCRIPTION
**Pull request by ickk**
While preparing #538 I needed to test the changes locally. Finding your way through the build process feels like *maze of twisty little passages, all alike*, which could be frustrating for new contributors. 

I roughly documented the process, but it could probably be generalised and made more concrete. We could even make a couple of simple scripts that automate setting up an environment similar to ci to ease the process.

### CODE REVIEW

File: local_build_instructions.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+### Windows local build instructions ###
+
+Building bevy-website locally is tedious, especially on windows. Here is one (brute force) method:
```
Comment (alice-i-cecile) [inline local_build_instructions.md:3]:
```suggestion
Building bevy-website locally is tedious, especially on Windows. Here is one (brute force) method:
```

File: local_build_instructions.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+### Windows local build instructions ###
```
Comment (alice-i-cecile) [inline local_build_instructions.md:1]:
```suggestion
# Windows local build instructions
```

**Comment by alice-i-cecile** [comment]
This lines up with my experience building this on Windows.

**Comment by IceSentry** [comment]
Personally, I prefer using git bash because it comes with git, so no additional install step and it uses your already existing installation of rust.

File: local_build_instructions.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+2. Open a WSL bash shell
+3. [install rustup](https://rustup.rs/) `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
+4. Add the wasm target `rustup target add wasm32-unknown-unknown`
+5. Install build dependencies `sudo apt install build-essential pkg-config`
+6. Install wasm-bindgen tool `cargo install wasm-bindgen-cli`
+
+Now we have everything we need to start generating the site. We can roughly follow the
+outline of the CI build script `.github/workflows/ci.ympl`
+
+7. cd into the website repo. There are a bunch of directories starting with "generate-".
```
Comment (IceSentry) [inline local_build_instructions.md:18]: I would add that these steps are optional if you don't plan to work on those related pages.

Comment (IceSentry) [reply local_build_instructions.md:18]: Like, when I was working on the asset page, I never generated all the other pages.

Comment (IceSentry) [reply local_build_instructions.md:18]: Thinking about it, we should probably just provide a `generate.sh` at the root that does all those steps

**Comment by mockersf** [comment]
I would prefer to discourage people from building parts they don't need as they are optional.

Another interesting option to explore could be to have the deploy process save everything generated to a branch that can be used when people need all the parts .

File: local_build_instructions.md (hunk 3 above)
Comment (ickk) [reply local_build_instructions.md:18]: Agree with you on git-bash being more minimal.. But I'm also not going to install a whole rust environment to git-bash when I already use WSL. I think there's value in having both.

I was hoping we could either flesh this out to include the other platforms (which will be pretty similar to the bulk, but I can't test mac), or just provide an automation for setting up the environment.

**Comment by ickk** [comment]
As for each thing being optional, I thought it would be pretty obvious for anyone wanting to contribute, but maybe some more explicit language would be good.

Mostly I just wanted to document what I did so next time I don't have to rediscover missing dependencies step by step. Feel free to change the wording, add stuff, whatever.

**Comment by ickk** [comment]
> Another interesting option to explore could be to have the deploy process save everything generated to a branch or an artefact to download that can be used when people need all the parts.

This could be useful, does sound maybe a little convoluted. Even so, I think regular documentation is important for when a contributor needs to change something that impacts the generated assets themselves.

**Comment by cart** [comment]
Maybe instead of going further down this path we should use cargo/xshell instead of bash for facilitating these builds (like we do for `cargo run -p ci`?

We should probably have a cargo workspace setup for this anyway.

generate_wasm_examples will be the biggest challenge. the `find` and `sed` usage specifically

**Pull request closed without merging**